vespa/deployment.py

```diff
@@ -778,7 +942,62 @@ def _set_target_cloud(self):
         else:
             print(output.stdout.decode("utf-8"))
 
+    def _vespa_auth_login(self):
+        is_notebook = is_jupyter_notebook()
+        # Open a new pseudo-terminal
+        master, slave = pty.openpty()
+
+        # Start the subprocess with its input/output connected to the PTY
+        p = subprocess.Popen(
+            shlex.split("vespa auth login"),
+            stdin=slave,
+            stdout=slave,
+            stderr=slave,
+            universal_newlines=True,
+        )
+
+        # Close the slave end in the parent process
+        os.close(slave)
+        finished = False
+        try:
+            while not finished:
+                # Use select to wait for data to be available on the PTY
+                rlist, _, _ = select.select([master], [], [], 1)
+
+                for fd in rlist:
+                    if fd == master:
+                        # Read output from the master end of the PTY
+                        output = os.read(master, 1024).decode("utf-8")
+                        if output:
+                            print(output, end="")
+                            sys.stdout.flush()
+                        if "Success:" in output:
+                            finished = True  # Exit the loop after success message
+                            break
+
+                        # Check for input only if running in a Jupyter Notebook
```

Comment:
What happens if the CLI requests input, but none is given?

Reply:
I'm just trying to understand the context. So if it's in a notebook (only), you answer "yes" for it? And otherwise, it's probably a user looking at it, that _can_ provide input?

Reply:
No, the thing is that if it is run from a python shell the `vespa auth login` will run and behave similar to the CLI  directly, but if running from a Jupyter Notebook or Colab, it will not ask the user for input (or submit the input) without this.

Reply:
So this is a hack to explicitly trigger a user input (when prompt is detected) and forward that input to the process.